backend/tracim_backend/lib/core/user.py

```diff
@@ -670,12 +681,66 @@ def authenticate(self, password: str, login: str, ldap_connector: "Connector" =
 
         raise AuthenticationFailed("Auth mechanism for this user is not activated")
 
+    def saml_authenticate(
+        self,
+        user: typing.Optional[User],
+        user_id: str,
+        name: str,
+        mail: str,
+    ) -> User:
+        """
+        Authenticate with ldap, return authenticated user or raise Exception
+        like WrongAuthTypeForUser, WrongLDAPCredentials, UserDoesNotExist
+        or UserAuthenticatedIsNotActive
+        :param user: user to check,, can be none if user not found, will try
+         to create new user if none but ldap auth succeed
+        :param user_id: user_id of the user
+        :param name: name extracted from saml attributes
+        :param mail: mail extracted from saml attributes
+        """
+        auth_type = AuthType.SAML
+
+        # TODO - M.L - 2023/09/06 - Change this to get saml response and isolate attributes
```

Comment:
What is this TODO @Millefeuille42 ? Is it still up-to-date?

Reply:
Nope, I will remove it

Reply:
Will remove it in #6258 (and will add a checkbox for it)

Reply:
removed in https://github.com/tracim/tracim/pull/6259